transitions/core.py

```diff
@@ -334,6 +337,7 @@ def __init__(self, model=None, states=None, initial=None, transitions=None,
                 executed in a state callback function will be queued and executed later.
                 Due to the nature of the queued processing, all transitions will
                 _always_ return True since conditional checks cannot be conducted at queueing time.
+            add_self (boolean): If no model(s) provided, intialize state machine against self.
```

Comment:
Not sure I like the idea of adding an `add_self` argument. I agree that we don't want to assume the user wants to add `self` to the model list automatically, but I think I'd prefer to handle this by making the default value of `None` leave the model list empty, and requiring a subsequent `add_model()` call. We could have a special case for the string value `self`--i.e., `Machine(model='self')` would produce the current behavior.

This isn't ideal, obviously, because it would break the API. But once we allow dynamic addition of models, it starts to seem quite counterintuitive to me to that the way to obtain a truly empty model list is to explicitly pass `add_self=False`. It seems like the more sensible approach is to default to an empty list, and make the user explicitly indicate if they want to populate the model list with the Machine instance itself.

Maybe we can keep the proposed approach for now, and save the API-breaking change for a major version release. Thoughts?